l2 announcer panic

### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

On a test cluster running Cilium 'dev' (as of 2023/06/08 using helm chart) and the bleeding-edge feature L2 announcement, and after having worked like a charm for a while, all Cilium pods crashed and stay in a CrashLoopBackOff state. No clue about what could have been the trigger.


Logs show a panic in l2announcer.





### Cilium Version

cilium-cli: v0.14.6 compiled with go1.20.4 on linux/amd64
cilium image (default): v1.13.3
cilium image (stable): v1.13.3
cilium image (running): -ci:latest   (as of 2023/06/08 using helm chart) 

### Kernel Version

Debian's 5.10.0-23-amd64

### Kubernetes Version

v1.27.2

### Sysdump

_No response_

### Relevant log output

```shell
level=info msg="Waiting until all Cilium CRDs are available" subsys=k8s
level=info msg="attempting to acquire leader lease kube-system/cilium-l2announce-minio-operator-operator..." subsys=klog
level=info msg="attempting to acquire leader lease kube-system/cilium-l2announce-test12-kibana-kb-http..." subsys=klog
panic: assignment to entry in nil map

goroutine 351 [running]:
github.com/cilium/cilium/pkg/l2announcer.svcAndMetaLabels(...)
        /go/src/github.com/cilium/cilium/pkg/l2announcer/l2announcer.go:1025
github.com/cilium/cilium/pkg/l2announcer.(*L2Announcer).upsertPolicy(0xc0012c45a0, {0x3b2e480, 0xc00118bc50}, 0xc001e10000)
        /go/src/github.com/cilium/cilium/pkg/l2announcer/l2announcer.go:501 +0xd50
github.com/cilium/cilium/pkg/l2announcer.(*L2Announcer).upsertLocalNode(0xc0012c45a0, {0x3b2e480, 0xc00118bc50}, 0x3b5ae40?)
        /go/src/github.com/cilium/cilium/pkg/l2announcer/l2announcer.go:819 +0x514
github.com/cilium/cilium/pkg/l2announcer.(*L2Announcer).processLocalNodeEvent(0x771b65?, {0x3b2e480?, 0xc00118bc50?}, {{0x3592c2c, 0x6}, {{0xc000715c86, 0xa}, {0x0, 0x0}}, 0xc0015ed000, ...})
        /go/src/github.com/cilium/cilium/pkg/l2announcer/l2announcer.go:768 +0x58
github.com/cilium/cilium/pkg/l2announcer.(*L2Announcer).run(0xc0012c45a0, {0x3b2e480, 0xc00118bc50})
        /go/src/github.com/cilium/cilium/pkg/l2announcer/l2announcer.go:176 +0x2c6
github.com/cilium/cilium/pkg/hive/job.(*jobOneShot).start(0xc00095db30, {0x3b2e480, 0xc00118bc50}, 0x6973756c63784510?, {{{0xc0018cd8c0, 0x1, 0x1}}, {0x3b5e7f0, 0xc0003cde30}, {0x3af1060, ...}})
        /go/src/github.com/cilium/cilium/pkg/hive/job/job.go:267 +0x530
created by github.com/cilium/cilium/pkg/hive/job.(*group).Start.func1
        /go/src/github.com/cilium/cilium/pkg/hive/job/job.go:137 +0x158
```


### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/pkg/l2announcer/l2announcer.go
+++ b/pkg/l2announcer/l2announcer.go
@@ -1022,6 +1022,10 @@ const (
 
 func svcAndMetaLabels(svc *slim_corev1.Service) labels.Set {
 	labels := maps.Clone(svc.GetLabels())
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+
 	labels[serviceNamespaceLabel] = svc.Namespace
 	labels[serviceNameLabel] = svc.Name
 	return labels

